stdlib/universe/contains.go

```diff
@@ -0,0 +1,104 @@
+package universe
+
+import (
+	"fmt"
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/semantic"
+	"github.com/influxdata/flux/values"
+	"github.com/pkg/errors"
+	"regexp"
+)
+
+// Contains will test whether a given value is a member of the given set array.
+type ContainsFunc struct{}
+
+func (c *ContainsFunc) Type() semantic.Type {
+	return semantic.Invalid
+}
+func (c *ContainsFunc) PolyType() semantic.PolyType {
+	return semantic.NewFunctionPolyType(semantic.FunctionPolySignature{
+		Parameters: map[string]semantic.PolyType{
+			"value": semantic.Tvar(1),
+			"set":   semantic.NewArrayPolyType(semantic.Tvar(2)),
+		},
+		Required: semantic.LabelSet{"value", "set"},
+		Return:   semantic.Bool,
+	})
+}
+func (c *ContainsFunc) IsNull() bool {
+	return false
+}
+func (c *ContainsFunc) Str() string {
+	panic(values.UnexpectedKind(semantic.Function, semantic.String))
+}
+func (c *ContainsFunc) Int() int64 {
+	panic(values.UnexpectedKind(semantic.Function, semantic.Int))
+}
+func (c *ContainsFunc) UInt() uint64 {
+	panic(values.UnexpectedKind(semantic.Function, semantic.UInt))
+}
+func (c *ContainsFunc) Float() float64 {
+	panic(values.UnexpectedKind(semantic.Function, semantic.Float))
+}
+func (c *ContainsFunc) Bool() bool {
+	panic(values.UnexpectedKind(semantic.Function, semantic.Bool))
+}
+func (c *ContainsFunc) Time() values.Time {
+	panic(values.UnexpectedKind(semantic.Float, semantic.Time))
+}
+func (c *ContainsFunc) Duration() values.Duration {
+	panic(values.UnexpectedKind(semantic.Float, semantic.Duration))
+}
+func (c *ContainsFunc) Regexp() *regexp.Regexp {
+	panic(values.UnexpectedKind(semantic.Float, semantic.Regexp))
+}
+func (c *ContainsFunc) Array() values.Array {
+	panic(values.UnexpectedKind(semantic.Float, semantic.Array))
+}
+func (c *ContainsFunc) Object() values.Object {
+	panic(values.UnexpectedKind(semantic.Float, semantic.Object))
+}
+func (c *ContainsFunc) Function() values.Function {
+	return c
+}
+func (c *ContainsFunc) Equal(rhs values.Value) bool {
+	f, ok := rhs.(*ContainsFunc)
+	return ok && (c == f)
+}
+func (c *ContainsFunc) HasSideEffect() bool {
+	return false
+}
+
+func (c *ContainsFunc) Call(args values.Object) (values.Value, error) {
+	v, ok := args.Get("value")
+	if !ok {
+		return nil, errors.New("missing argument value")
+	}
+
+	setarg, ok := args.Get("set")
+	if !ok {
+		return nil, errors.New("missing argument set")
+	}
+
+	set := setarg.Array()
+	found := false
+	var err error
+	if set.Len() > 0 {
+		if set.Get(0).Type() != v.Type() {
+			err = fmt.Errorf("value type %T does not match set type %T", v.Type(), set.Get(0).Type())
+		} else {
+			for i := 0; i < set.Len(); i++ {
+				if set.Get(i).Equal(v) {
+					found = true
+					break
+				}
+			}
+		}
+	}
+
+	return values.NewBool(found), err
+}
+
+func init() {
+	flux.RegisterPackageValue("universe", "contains", &ContainsFunc{})
```

Comment:
Can you use `values.NewFunction` instead? Technically `values.NewFunction` require monotypes but that is only because it hasn't yet been updated. It should be straightforward to update it to use poly types.